### DISCUSSION TITLE
Is endpoint cpm accurate?

### DISCUSSION THREAD

**Discussion by JHub-Wei** [apache/skywalking#7484] [Q&A] [answered]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ * ] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?
We perform a pressure test on an endpoint at 1000 TPS. However, the CPM value of the endpoint is only 7438 on the dashboard during this period. 7438 / 60 = 124,it doesn't match the actual tps.Why is this happening?

The test service is a single instance. The OAP cluster is in cluster mode and ZooKeeper is used.
___
### Bug
- Which version of SkyWalking, OS, and JRE?
8.4.0，jdk1.8
- Which company or project?

- What happened?
If possible, provide a way to reproduce the error. e.g. demo application, component version.

___
### Requirement or improvement
- Please describe your requirements or improvement suggestions.

**Comment by wu-sheng** [comment, marked as answer]
If OAP is running in the cluster mode with ZooKeeper, you need to make sure their binding-IPs are not `0.0.0.0`. Otherwise, you are going to face dirty write easily.

**Reply by wu-sheng** [reply]
Quote answer from https://github.com/apache/skywalking/discussions/7484#discussioncomment-1199577

___
It depends. Could be SW_CORE_GRPC_HOST, or internalComHost or internalComPort, read document before you take actions, https://skywalking.apache.org/docs/main/latest/en/setup/backend/backend-cluster/#zookeeper-coordinator
___

**Comment by JHub-Wei** [comment]
Of course, their binding-IPs are not 0.0.0.0.The environment variable SW_CLUSTER_ZK_HOST_PORT is configured.

**Reply by wu-sheng** [reply]
`SW_CLUSTER_ZK_HOST_PORT` is for ZooKeeper server address. I am asking for OAP binding-IPs.

**Reply by JHub-Wei** [reply]
What Does OAP Binding-IPs Mean?Is the environment variable SW_CORE_GRPC_HOST?

**Reply by wu-sheng** [reply]
It depends. Could be `SW_CORE_GRPC_HOST`, or `internalComHost` or `internalComPort`, read document before you take actions, https://skywalking.apache.org/docs/main/latest/en/setup/backend/backend-cluster/#zookeeper-coordinator

**Reply by JHub-Wei** [reply]
Thx，I'll take a look first.

**Comment by JHub-Wei** [comment]
If the OAP is not in cluster mode before, do I need to restart the service integrated with the agent after the OAP is changed to cluster mode?

**Reply by wu-sheng** [reply]
No, you don't need to reboot agent.

**Comment by larrying** [comment]
did you resolve this problem? i faced the same problem

**Reply by JHub-Wei** [reply]
Yes, We set SW_CORE_GRPC_HOST in deployment, and it works.
            - name: SW_CORE_GRPC_HOST
              valueFrom:
                fieldRef:      
                  fieldPath: status.podIP

**Comment by larrying** [comment]
thank you, it works
